src/index.ts

```diff
@@ -589,6 +561,61 @@ class Envoy {
       return new Package(true, { "type": "success" }, { response_id });
     }, this.opts.NS);
   }
+
+  async signIn(hha_hash, agent_id): Promise<void> {
+    if (agent_id in this.anonymous_agents) {
+      // Nothing to do. Anonymous cell is always active
+      return;
+    }
+
+    const hosted_agent_instance_app_id = `${hha_hash}:${agent_id}`;
+
+    // Activate App - Tell Holochain to begin gossiping and be ready for zome calls on this app.
+    try {
+      log.info("Activating Installed App (%s)", hosted_agent_instance_app_id);
+      const adminResponse = await this.callConductor("admin", 'activateApp', { installed_app_id: hosted_agent_instance_app_id });
+
+      if (adminResponse.type !== "success") {
+        log.error("Conductor 'activateApp' returned non-success response: %s", adminResponse);
+        throw (new HoloError(`Failed to complete 'activateApp' for installed_app_id'${hosted_agent_instance_app_id}'.`)).toJSON();
+      }
+    } catch (err) {
+      if (err.message.includes("Tried to activate an app that was not installed")) {
+        // This error is returned in two cases:
+        // a) The app is not installed -- Return an error to the user saying that they may need to sign up first.
```

Comment:
Are we sure if it is "not installed" that holochain `throws` an error and not just `returns` an error?

Reply:
I know that holochain conductor API throws an error if the response is of the `Error` enum variant. https://github.com/holochain/holochain-conductor-api/blob/7519b31134e848d51e93e4b860d9f0d31f7d0b3b/src/websocket/common.ts#L1-L8

 I don't think our wrapper ever catches that error but if it didn't catch errors it seems very weird that it bothers adding `.type = "success"` to the return value.